### DISCUSSION TITLE
Running Roslyn analyzer with file change tracking outside IDE

### DISCUSSION THREAD

**Discussion by ever-modern** [dotnet/roslyn#69491] [Q&A]
I would like to make a program that will subscribe for project files content change and yield new Compilation object for all those changes. Getting the changed syntax nodes is desirable too. Probably this all will be happening the way it does in VS.
Is there a simple example of how to start all that process?

**Comment by miyu** [comment]
See also https://github.com/dotnet/roslyn/issues/7842
